wallet/src/de/schildbach/wallet/data/UsernameSearchResult.kt

```diff
@@ -1,30 +1,45 @@
 package de.schildbach.wallet.data
 
-import kotlin.math.max
-
 data class UsernameSearchResult(val username: String,
                                 val dashPayProfile: DashPayProfile,
                                 var toContactRequest: DashPayContactRequest?,
                                 var fromContactRequest: DashPayContactRequest?) {
     val requestSent: Boolean
-            get() = toContactRequest != null
+        get() = toContactRequest != null
     val requestReceived: Boolean
-            get () = fromContactRequest != null
+        get() = fromContactRequest != null
     val isPendingRequest: Boolean
-            get() = requestReceived && !requestSent
+        get() = requestReceived && !requestSent
+
     val date: Long // milliseconds
-            get() {
-                return when (requestSent to requestReceived) {
-                    true to true -> {
-                        fromContactRequest!!.timestamp
-                    }
-                    true to false -> {
-                        toContactRequest!!.timestamp
-                    }
-                    false to true -> {
-                        fromContactRequest!!.timestamp
-                    }
-                    else -> 0.00
-                }.toLong() * 1000
-            }
+        get() {
+            return when (type) {
+                Type.CONTACT_ESTABLISHED -> {
+                    fromContactRequest!!.timestamp
+                }
+                Type.REQUEST_SENT -> {
+                    toContactRequest!!.timestamp
+                }
+                Type.REQUEST_RECEIVED -> {
+                    fromContactRequest!!.timestamp
+                }
+            }.toLong() * 1000
+        }
+
+    val type: Type
+        get() = when (requestSent to requestReceived) {
+            true to true -> Type.CONTACT_ESTABLISHED
+            false to true -> Type.REQUEST_RECEIVED
+            true to false -> Type.REQUEST_SENT
+            else -> throw IllegalStateException("toContactRequest and fromContactRequest can't both be null at the same time")
+        }
+
+    val incoming: Boolean
+        get() = toContactRequest!!.timestamp > fromContactRequest!!.timestamp
```

Comment:
Here, incoming appears to mean that we received a request first, then we sent our request later.

One issue here is that a UserSearchResult is also used for pending contact requests, in which cases `toContactRequest` will be null.  The NotificationActivity and ContactsFragment.  At this time, I don't think `incoming` will be used on either screen.

For the DashPayUserActivity screen, currently, or at least the way I wrote it, the history is only shown when there is a contact request going both ways (neither toContactRequest or fromContactRequest is null).

Also, I think the SearchUserActivity screen also uses UsernameSearchResult to hold a sent request where fromContactRequest is null.  While we don't use `incoming` in that activity, maybe it is better to make this getter null pointer exception proof.

Reply:
I replaced it with a local variable inside `NotificationsForUserLiveData.searchNotifications`